ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/websocket/subscription/SubscriptionManager.java

```diff
@@ -160,4 +161,22 @@ public void sendMessage(final Long subscriptionId, final JsonRpcResult msg) {
         .findFirst()
         .ifPresent(connectionId -> vertx.eventBus().send(connectionId, Json.encode(response)));
   }
+
+  public <T> Void notifySubscribersOnWorkerThread(
+      final SubscriptionType subscriptionType,
+      final Class<T> clazz,
+      final Consumer<List<T>> runnable) {
+    vertx.executeBlocking(
+        future -> {
+          final List<T> syncingSubscriptions = subscriptionsOfType(subscriptionType, clazz);
+          runnable.accept(syncingSubscriptions);
+          future.complete();
+        },
+        result -> {
+          if (result.failed()) {
+            LOG.error("Failed to notify subscribers. Cause: {}", result.cause().getMessage());
```

Comment:
Rather than logging just the message, include the full stack trace.
```suggestion
            LOG.error("Failed to notify subscribers.", result.cause());
```